docs: Update overview.md (#1285)
aave/aave-v4: docs/overview.md

Reviewer: ```suggestion
Premium Debt is the portion of a user’s debt that represents the additional interest accumulated due to the quality of the user’s collateral assets (i.e., their Risk Premium on top of the base drawn rate).
```

Reviewer: ```suggestion
3. The snapshot does **not** update on actions that do not increase the risk exposure of the system:
```

Aave V4 introduces an architectural redesign centered around the Hub, enabling protocol flexibility and capital efficiency. This innovative architecture allows the Governor (e.g., the Aave DAO) to dynamically manage Spokes, adding new borrowing capabilities and removing outdated ones without requiring costly liquidity migrations.

> The protocol implements sophisticated risk management through its Risk Premiums system, where each asset receives a specific risk factor (called Collateral Risk) ranging from 0 to `1000_00` (BPS) based on the asset's implied volatility, market conditions, liquidity, risk, etc. Base drawn rates are determined purely by utilisation, while risk premium rates are adjusted according to collateral risk profiles, from low‑risk pristine‑quality assets (such as ETH) to higher‑risk collateral, scaling borrowing costs proportionally.

Reviewer: I think it reads better this way, and also use back ticks for all numbers
```suggestion
The protocol implements sophisticated risk management through its Risk Premium system, where each asset receives a specific risk factor (called Collateral Risk) ranging from `0` to `1000_00` (BPS) based on the asset's implied volatility, market conditions, liquidity, risk, etc. Base drawn rates are determined purely by utilisation, while risk premium rates are adjusted according to collateral risk profiles, from low‑risk pristine‑quality assets (such as ETH) to higher‑risk collateral, scaling borrowing costs proportionally.
```

- Managing interest rate strategy per asset, defining the optimal usage ratio, base drawn rate, and rate slopes, among other parameters.
- Managing the access controls for the Governor entity authorized to execute emergency controls, the authorized entity which can call `mintFeeShares`, and general access controls via the `AccessManager` authority.
- Setting the liquidity fee per asset, determining the share of interest revenue retained by the protocol.
- Enforcing accounting and solvency invariants:
  1. Hub level aggregate fields == sum across all per-Spoke fields (`addedShares`, `drawnShares`, `premiumShares`, `premiumOffsetRay`, `deficitRay`)
  2. Supply share price and drawn index cannot decrease (remains constant or increases)

>   3. Hub's actual token balance for a given asset is always >= internally tracked available liquidity (`asset.liquidity`)

Reviewer 2: pls review updated hub invariants

- Employing reentrancy guards for extra protection against reentrancy attacks, even though the Hub, Interest Rate Strategy, and Price Feeds are trusted and Aave V4 does not support tokens with callbacks.
- Enforcing position constraints through a configurable `MAX_USER_RESERVES_LIMIT` which limits the number of collateral Reserves and the number of borrow Reserves a user can have (each counted separately).
- Configuring and enforcing per-reserve liquidation parameters at the Spoke level.
- Enforcing position safety invariants:
  1. Users without collateral cannot assume debt

>   2. No user action can worsen their position's health factor below the liquidation threshold

Reviewer 2: pls review updated spoke invariants

- **Governance‑Controlled**: The Governor manages all funds allocated to reinvestment strategies. Decisions on strategy selection, risk parameters, and operational guidelines are exclusively governance responsibilities.
- **Interest Rate Neutral**: Swept liquidity remains part of the usage ratio denominator, so sweeping funds to external strategies does not affect the drawn rate experienced by users.

> - **Opt‑In for Users**: Participation in reinvestment is not compulsory. Reinvestment is enabled at the Hub level on a per-asset basis; users choose whether to interact with a reinvestment-enabled asset by supplying to a spoke Reserve that is connected to that Hub asset. Since the reinvestment controller is a governance-controlled parameter that can be activated at any time. It is the Governor's responsibility to properly communicate any planned activation and its timeframe in advance, giving existing suppliers the opportunity to withdraw their funds if they wish to opt out.

Reviewer 2: ```suggestion
- **Opt‑In for Users**: Participation in reinvestment is not compulsory. Reinvestment is enabled at the Hub level on a per-asset basis; users choose whether to interact with a reinvestment-enabled asset by supplying to a spoke Reserve that is connected to that Hub asset. Since the reinvestment controller is a governance-controlled parameter that can be activated at any time, it is the Governor's responsibility to properly communicate any planned activation and its timeframe in advance. This gives existing suppliers the opportunity to withdraw their funds if they wish to opt out.
```

> - **New Yield Opportunities**: By deploying otherwise idle liquidity into external strategies, opt‑in liquidity providers can potentially earn incremental returns on top of borrower interest.

Author: I'll change the first part but keep "potentially." Otherwise, it looks like additional earnings are guaranteed. In the next bullet point, it clarifies that any losses are absorbed by the Governor, removing doubts about this situation

- **Risk Allocation**: Losses arising from reinvestment strategies are absorbed by the Governor, who defines the coverage method, shielding liquidity providers from direct strategy‑specific risk exposure.
- **Optional by Design**: The module can remain disabled without affecting core supply and borrow functionality. When inactive, all liquidity stays in the Hub.

The reinvestment infrastructure enables the protocol to optimize capital efficiency while maintaining user choice and clear risk allocation. Implementation details, strategy selection, and risk management frameworks are governance decisions that can evolve based on protocol needs and market conditions.

# Dynamic Risk Configuration

> One of the major risk‑side limitations of Aave V3 lies in its single, global risk configuration per asset. This design creates significant governance overhead and potential user harm through unexpected liquidations, as any parameter change, in particular lowering the liquidation threshold, immediately affects every open position.

Author: I see your point now, moving forward with it

## Key Differences from Aave V3

- **Target Health Factor vs Close Factor:** In Aave V3, the default close factor is 50% (with a 100% close factor when HF < 0.95 or when liquidation amounts are under a given base currency threshold). Liquidators would typically repay half of a borrower’s debt and seize half of their collateral. Aave V4 removes the default close‑factor: the maximum a liquidator can repay (in situations that do not result in dust collateral or debt remaining) is the amount needed to bring the borrower back to the `TargetHealthFactor` determined by the Governor.
- **Dynamic Dust Handling during Liquidations**: Both Aave V3 and Aave V4 revert when the remaining amount is below a hard‑coded threshold, while dynamically adjusting the maximum debt that can be liquidated and, if the liquidator opts to fully repay, allow full repayment to prevent dust. However, Aave V4 allows more flexibility because of removing the close-factor and facilitating the liquidation steps required to bring the position back to the target HF. Dust may still remain if either the collateral or debt Reserve is fully liquidated.

> - **Dutch‑Auction Style Liquidation Bonus:** Aave V3 applies a static liquidation bonus that does not depend on the borrower’s health factor. Aave V4 introduces a variable liquidation bonus that increases linearly as the health factor decreases. Governance can specify two spoke‑wide parameters that shape the liquidation bonus: `healthFactorForMaxBonus` and `liquidationBonusFactor`.

Reviewer: I'd be fine with "shape" (more flavor text), but ok

5. **Apply Debt Repayment & Transfer Collateral**: Reduce the borrower’s debt amount by the repaid amount. Transfer the corresponding collateral to the liquidator with the liquidation bonus applied, minus the protocol fee (as in Aave V3). If `receiveSharesEnabled` is true for the collateral Reserve and the Reserve is not frozen, the liquidator can opt to receive added shares directly instead of underlying assets by setting the `receiveShares` parameter to true. Added shares accrue yield in the Hub, providing a more capital-efficient liquidation mechanism. The fee portion is sent to the protocol/fee receiver via the Hub as shares, accrues yield there, and the shares are assigned directly via Hub accounting.

> 6. **Emit Events and Update State:** A `LiquidationCall` event is emitted containing details of the liquidation. The borrower’s and Reserve’s interest indices are updated. If the borrower still has debt outstanding and no remaining collateral, the system will record a protocol deficit. Reporting deficit is allowed even when the reporting Spoke is halted (as long as it’s active).

Reviewer: There is no borrower (user) interest index, can remove that part. Or more accurate to just say 
```suggestion
6. **Emit Events and Update State:** A `LiquidationCall` event is emitted containing details of the liquidation. The borrower’s and Reserve’s debt amounts are updated. If the borrower still has debt outstanding and no remaining collateral, the system will record a protocol deficit. Reporting deficit is allowed even when the reporting Spoke is halted (as long as it’s active).
```
